### ISSUE TITLE
[Feature Request] Support for arbitrary numbers of multiple captions

### ISSUE THREAD

**Issue by hrothgar234567** [Hysocs/Aozora_SDXL_Training#9]
Historically, tools like sd-scripts have supported caption augmentations such as tag shuffling, tag dropout, etc. It is my understanding that these augmentations are primarily useful for LoRA training so as to avoid overfitting a small dataset, and thus are not imminently useful for a full finetune. I am trying to find ways to enhance models' ability to distinguish between highly correlated tags (the most notable examples being character tags being highly correlated with tags which describe their appearance). I generated a tag coincidence matrix for all pairs of tags across the danbooru 2026 dataset. Using this, I was able to find lists of tags highly correlated with each character tag. In testing with LoRA, I've had good success with randomly sampling a captions from the set of `character tag without highly correlated tags` and `no character tag but including descriptive tags`, and I'd like to see if that success transfers to a full finetune. In short, would it be possible to support lists of captions for each category in your json format and have it randomly select a caption each time an image is selected for training? I assume this would require changes in how you cache the TE embeddings of the captions.

**Comment by Hysocs** [comment]
Instead of me changing the trainer code, you could make a script that goes through your images and pre-generates a caption for each one in whatever format you want. The trainer would then just use those captions.

If you want two separate tags for the same image, you could include the image twice in the dataset with a different caption each time. It is not really the proper way to do it, but it should work well enough for testing.
